### ISSUE TITLE
Build failed for labwc-git with latest wlroots-git

### ISSUE THREAD

**Issue by EysseW** [labwc/labwc#415]
Non buch of a bug but I need help.
[build-failed.txt](https://github.com/labwc/labwc/files/8971235/build-failed.txt)

**Comment by EysseW** [comment]
Using wlroot-git and i'm trying to install labwc-git

**Comment by Consolatis** [comment]
Look like we have some wlroots chasing to do again.
For now you could work around it like this:
```bash
git clone https://github.com/labwc/labwc
cd labwc
meson build --force-fallback-for=wlroots
meson compile -C build
build/labwc
```

Makes me wonder if the `labwc-git` package for AUR could set that as default build option and remove the dependency to `wlroots-git` because this will happen over and over again until wlroots does a release which could then be targeted instead of `wlroots-git`.

@narrat thoughts?

**Comment by EysseW** [comment]
Thanks! I actually really like that it is using wlroots-git because so is sway-borders-git and hyprland. I wanna use both of these packages really bad.

**Comment by Consolatis** [comment]
AFAIR we are linking wlroots statically when using the fallback so having a different system wide installation of `wlroots-git` around should be fine.

**Comment by Narrat** [comment]
The prefered way is how it is currently done.
But if breaking changes are that common and if there is no chance for conflicting files (especially with wlroots-git) I can switch it to the fallback way of building labwc.

**Comment by Consolatis** [comment]
> But if breaking changes are that common

They are actually not \*all\* that common but hard to predict.
Can be fine for some weeks / months but then break 5 times in a row because of upstream refactorings.

Should be fixed for now once #417 is merged but may break again in the future.

**Comment by Narrat** [comment]
I'm not sure what to think of such a change. In general -git packages and the like are only advised for devs and they should know how to handle those cases.
At least `wlroots` and `wlroots-git` won't ship the libwlroots.a file. But I would need to remove the include dir.
And add some additional deps as duplicates to make sure it will work after installing it. 
Personally I wouldn't change a thing and live with random breakages, as those are getting resolved.

Diff for the those who want to change it (still missing the additional depends and doesn't handle conflicting files):
```diff
diff --git a/PKGBUILD b/PKGBUILD
index d8724dc..1234e94 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -2,14 +2,15 @@
 
 _pkgname=labwc
 pkgname=labwc-git
-pkgver=0.5.0.r76.g5fd5024
-pkgrel=2
+pkgver=0.5.0.r230.g437230b
+pkgrel=1
 pkgdesc='stacking wayland compositor with look and feel from openbox (git version)'
 url="https://github.com/labwc/labwc"
 arch=('x86_64')
 license=('GPL2')
-depends=('pango' 'seatd' 'wlroots>=0.16' 'wlroots<0.17' 'wayland' 'xorg-xwayland')
+depends=('pango' 'seatd' 'wayland' 'xorg-xwayland')
 makedepends=('git' 'meson' 'scdoc' 'wayland-protocols')
+makedepends+=('vulkan-headers' 'xorgproto') # for latest compatible wlroots
 optdepends=("bemenu: default launcher via Alt+F3")
 conflicts=(labwc)
 provides=(labwc)
@@ -22,8 +23,12 @@ pkgver() {
   git describe --long | sed 's/\([^-]*-g\)/r\1/;s/-/./g'
 }
 
+prepare() {
+  meson subprojects download wlroots --sourcedir="$_pkgname"
+}
+
 build() {
-  arch-meson -Dman-pages=enabled "$_pkgname" build
+  arch-meson -Dman-pages=enabled "$_pkgname" build --force-fallback-for=wlroots
   meson compile -C build
 }
 

```
